source/fundamentals/connection.txt

```diff
@@ -0,0 +1,435 @@
+-----------
+
+You can compress messages passing between your MongoDB instance and the
+driver. MongoDB drivers support up to three different algorithms
+depending on release version:
+
+1. `Snappy <https://google.github.io/snappy/>`__: available in MongoDB 3.4 and later.
+
+2. `Zlib <https://zlib.net/>`__: available in MongoDB 3.6 and later.
+
+3. `Zstandard <https://github.com/facebook/zstd/>`__: available in MongoDB 4.2 and later.
+
+You can specify one or more compression algorithms, but the driver uses
+the first compressor in the list supported by the connected
+MongoDB instance.
+
+.. _compression:
+
+Enable Compression
+~~~~~~~~~~~~~~~~~~
+
+You can enable compression for the connection to your MongoDB instance
+in two different ways: through a parameter in your connection string, or
+using a method in the ``MongoClientSettings.Builder`` class.
+
+.. tabs::
+
+   .. tab:: ConnectionString
+      :tabid: connectionstring
+
+      To enable compression with a :java-sync-api:`ConnectionString
+      <com/mongodb/ConnectionString.html>`, use the parameter ``compressors``
+      in the connection string passed to
+      ``MongoClients.create()``. You can specify one or more compression
+      algorithms, separating multiple algorithms with commas:
+
+      .. code-block:: java
+
+         ConnectionString connectionString = new ConnectionString("mongodb+srv://<user>:<password>@<cluster-url>?compressors=snappy,zlib,zstd");
+         MongoClient mongoClient = MongoClients.create(connectionString);
+
+      Specify compression algorithms using the following strings:
+
+      - "snappy" for `Snappy <https://google.github.io/snappy/>`__ compression.
+
+      - "zlib" for `Zlib <https://zlib.net/>`__ compression.
+
+      - "zstd" for `Zstandard <https://github.com/facebook/zstd/>`__ compression.
+
+   .. tab:: MongoClientSettings
+      :tabid: mongoclientsettings
+
+      To enable compression with :java-sync-api:`MongoClientSettings
+      <com/mongodb/MongoClientSettings.html>`, pass the
+      :java-sync-api:`compressorList()
+      <com/mongodb/MongoClientSettings.Builder.html#compressorList(java.util.List)>`
+      builder method a list of :java-sync-api:`MongoCompressor <com/mongodb/MongoCompressor.html>`
+      instances. You can specify one or more compression algorithms in the list:
+
+      .. code-block:: java
+         :emphasize-lines: 2-4
+
+         MongoClientSettings settings = MongoClientSettings.builder()
+              .compressorList(Arrays.asList(MongoCompressor.createSnappyCompressor(),
+                                            MongoCompressor.createZlibCompressor(),
+                                            MongoCompressor.createZstdCompressor()))
+              .build();
+         MongoClient client = MongoClients.create(settings);
+
+.. _compression-dependencies:
+
+Compression Algorithm Dependencies
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The JDK supports `Zlib <https://zlib.net/>`__ compression natively, but
```

Comment:
You should mention that applications that require snappy or zstd need to add their own explicit dependency on those libraries.